probe-rs-target/src/chip_family.rs

```diff
@@ -288,3 +389,44 @@ impl ChipFamily {
         self.flash_algorithms.iter().find(|elem| elem.name == name)
     }
 }
+
+/// An error while loading a chip family ocurred.
+#[derive(Debug, thiserror::Error)]
+pub enum LoadError {
+    #[error("target dir at {0:?} could not be read")]
+    TargetDirRead(PathBuf),
+    #[error("dir entry could not be read")]
+    DirEntryRead(#[source] std::io::Error),
+    #[error("target definition file at {0:?} could not be read")]
+    TargetDefinitionFileRead(PathBuf),
+    #[error("target definition file at {path:?} could not be parsed")]
+    TargetDefinitionFileParse {
+        path: PathBuf,
+        #[source]
+        parent: serde_yaml::Error,
+    },
+    #[error("target file at {0:?} is not a yaml")]
+    NotAYaml(PathBuf),
+    #[error("target file at {0:?} could not be read")]
+    TargetFileRead(PathBuf),
+    #[error("unrecognized target file at {0:?}")]
+    UnrecongnizedFile(PathBuf),
+    #[error("The yaml format did not match the expected one")]
+    UnexpectedYamlFormat(#[source] serde_yaml::Error),
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::ChipFamily;
+
+    #[test]
+    fn basic_merge() {
+        let values = vec![
+            serde_yaml::from_str(r#"{ a: 65, b: { b: 66 }, c: { c: 67 } }"#).unwrap(),
+            serde_yaml::from_str(r#"{ a: null, b: { a: 65 }, c: 67 }"#).unwrap(),
+        ];
+
+        let merged = ChipFamily::merge(values).unwrap();
+        insta::assert_yaml_snapshot!(merged);
```

Comment:
Does this have to be piped through insta? I wouldn't think comparing two 4-line yaml blobs really wants a solution where the expectation is in a different folder.

Reply:
I like it because it makes spotting differences really nice and manageable. It is a dev dependency, so it should not matter much.